### PR TITLE
ENH: add axis to sparse norms

### PR DESCRIPTION
**Pull request by argriffing**
This PR adds an `axis` keyword argument to sparse norms, using code from the numpy norm implementation and pieces of https://github.com/scipy/scipy/pull/3303.  The `keepdims` keyword argument is not yet implemented.

The sparse norm implementation in this PR returns scalars or 1d ndarrays.  This behavior is not standard for sparse matrices, and contributes to the following confusion:

``` python
>>> s = scipy.sparse.identity(4).tocsr()
>>> s.max(axis=0)
<1x4 sparse matrix of type '<type 'numpy.float64'>'
        with 4 stored elements in COOrdinate format>
>>> s.sum(axis=0)
matrix([[ 1.,  1.,  1.,  1.]])
>>> spnorm(s, axis=0)
array([ 1.,  1.,  1.,  1.])
```

### CODE REVIEW

**Comment by larsmans** [comment]
I wonder if making this a freestanding function is the right design. On `_data_matrix` subtypes, the Frobenius norm can be computed much more easily as ``norm(X.data)``, which suggests that methods may be more appropriate. I also don't really like the implicit conversion to CSR.

Note that scikit-learn has [Cython code CSR row norms](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/utils/sparsefuncs_fast.pyx#L20).

**Comment by argriffing** [comment]
> I wonder if making this a freestanding function is the right design.

Maybe it wasn't the right design, but `sparse.linalg.norm` has already been added as a freestanding function by some commits from the closed PR https://github.com/scipy/scipy/pull/3303 which have been merged in the rebase https://github.com/scipy/scipy/pull/4505 (the axis args were left behind in that rebase, and the point of this PR is to add them back).

Would it make sense to add a member function like `_norm` to be queried by the freestanding `norm` function? This could be added in a different PR, separate from the issue of adding `axis` args.

**Comment by perimosocordiae** [comment]
I'm +1 for handling the axis kwarg now and worrying about efficient implementations later. LGTM.

**Comment by pv** [comment]
LGTM. Let's worry about performance later. (=when someone complains on stackoverflow)
